tests/UserControllerTest.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+
+class UserControllerTest extends TestCase {
+    
+    public function testGetAction() {
+        $mock = $this->getMockBuilder("App\Controllers\UserController")
```

Comment:
1) Implement BaseController -> setService(BaseService $service) method, which will set $this->service
2) user BaseController -> setService to pass your service mock